bin/node/src/service.rs

```diff
@@ -252,16 +259,18 @@ fn setup(
             warp_sync: None,
         })?;
 
+    let chain_status = SubstrateChainStatus::new(backend.clone())
```

Comment:
nit: Maybe you can share that `SubstrateChainStatus` between `setup` and `new_authority`.  Probably that is not an issue here, but it's not great to share same error messages here and there.

Reply:
Hmmm, trueeee... This whole thing could use a refactor, but in this specific case we are making things strictly, and potentially noticeably, worse.

Moved the creation around a bit to make it only happen once; the price is `setup` now takes it as an argument.